Sources/AsyncHTTPClient/ConnectionPool/State Machine/HTTPConnectionPool+HTTP1StateMachine.swift

```diff
@@ -219,6 +223,17 @@ extension HTTPConnectionPool {
 
             switch self.lifecycleState {
             case .running:
+                guard self.retryConnectionEstablishment else {
```

Comment:
Instead of using the guard here it might be nicer to go with where clause in the switch...

Reply:
A where constraint only applies to the last case if multiple cases are defined e.g.
```swift
case .running, .shuttingDown where self.retryConnectionEstablishment == true:
```
the where constrain only applies to `.shuttingDown` but not `.running`. We had an issue in AHC which we debugged together and decided we should stop using where clauses in switch cases altogether because it is not obvious.

Reply:
true, however here we only have one case here. Anyway, not a hill I want to die on. If we want to stick with an inline if/guard, I would prefer this to be an `if self.retryConnectionEstablishment {}` and then move the retry logic into the if.

Reply:
What's the benefit of the `if`? Guard at least guarantees that we return, if can fall through which we don't want.

Reply:
I consider it nicer to read. But that's personal...